dbot/dbot/Program.cs

```diff
@@ -35,20 +40,34 @@ public async Task MainAsync(string[] args)
 
             var discordToken = tokenManager.GetToken(TokenKey.DiscordToken);
             var omdbToken = tokenManager.GetToken(TokenKey.OMDBToken);
-            var nominationsFile = tokenManager.GetToken(TokenKey.NominationsFile);
-            var votesFile = tokenManager.GetToken(TokenKey.VotesFile);
+         //   var nominationsFile = tokenManager.GetToken(TokenKey.NominationsFile);
+     //       var votesFile = tokenManager.GetToken(TokenKey.VotesFile);
             Console.WriteLine($"Hello World! {omdbToken} {discordToken}");
 
             client = new DiscordSocketClient();
             commands = new CommandService();
             serviceCollection = new ServiceCollection();
           
-            serviceCollection.AddSingleton(new NominationsService(new AutoSerializedDictionary<User, Nomination>(nominationsFile)));
-            serviceCollection.AddSingleton(new VotingService(new AutoSerializedDictionary<User, int>(votesFile)));
+            serviceCollection.AddDbContext<MovieBotContext>(options => options.UseSqlite(DBCONN));
+            serviceCollection.AddScoped<IDbManager,DbManager>();
+            serviceCollection.AddScoped<NominationsService>();
+            serviceCollection.AddScoped<VotingService>();
             serviceCollection.AddSingleton(new OmdbService(omdbToken));
             serviceCollection.AddSingleton(commands);
 
             services = serviceCollection.BuildServiceProvider();
+
+            //check for db and create it if it does not exist
+            try
+            {
+                var context = services.GetRequiredService<MovieBotContext>();
+                DbInitializer.Initialize(context);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Something went wrong during DB initialize: " + ex.Message);
```

Comment:
Would it make sense to crash here instead of logging and continuing?

Reply:
Yeah  I agree that crashing on db init would make the most sense. i can rethrow the exception to force the crash. Is there a place we want to record the exception messages? I dont think we wanted to keep writing to the console/debug console?

Reply:
For now we can just print that something went wrong and then exit gracefully I think. In the future we should consider adding the ability to log to a log file instead